src/smc-webapp/project_store.coffee

```diff
@@ -194,6 +194,13 @@ class ProjectActions extends Actions
                                 @_project().show_editor_chat_window(opts.path)
                             if opts.foreground
                                 @set_current_path(misc.path_split(opts.path).head, update_file_listing=false)
+        # log file opening to HTML5 storage so we can auto reopen tabs at some point
+        if (!(window.localStorage.getItem("opened_files")))
+            window.localStorage.setItem("opened_files", "[]")
+        opened_files = JSON.parse(window.localStorage.getItem("opened_files"))
+        console.log('opened_files:', window.localStorage.getItem("opened_files"))
+        if (!(window.localStorage.getItem("opened_files").indexOf(JSON.stringify({'project_id': @project_id, 'path': opts.path})) > -1))
+            opened_files.push({'project_id': @project_id, 'path': opts.path})
```

Comment:
But then you don't save opened_files...?  So you just pushed a bunch to it, which is thrown away?  I'm confused.